src/modules/billing/services/supplementary-billing-service/data-service.js

```diff
@@ -0,0 +1,138 @@
+'use strict';
+
+const { groupBy, omit } = require('lodash');
+
+// Models
+const Transaction = require('../../../../lib/models/transaction');
+
+// Services
+const billingTransactionsRepo = require('../../../../lib/connectors/repos/billing-transactions');
+const invoiceService = require('../invoice-service');
+const invoiceLicencesService = require('../invoice-licences-service');
+
+const { actions } = require('./constants');
+
+/**
+ * These are the keys present on a transaction object
+ * from the initial DB query and subsequent processing
+ * which must be omitted before attempting to write the
+ * transaction back to the DB
+ */
+const omitKeys = [
```

Comment:
Would it not be safer to just pick the data you need. What if later on someone adds another key to objects this might then fail?

Reply:
Agreed, I'll change this